routers/events/events.go

```diff
@@ -0,0 +1,113 @@
+// Copyright 2020 The Gitea Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package events
+
+import (
+	"net/http"
+	"time"
+
+	"code.gitea.io/gitea/modules/context"
+	"code.gitea.io/gitea/modules/eventsource"
+	"code.gitea.io/gitea/modules/graceful"
+	"code.gitea.io/gitea/modules/log"
+	"code.gitea.io/gitea/routers/user"
+)
+
+// Events listens for events
+func Events(ctx *context.Context) {
+	// FIXME: Need to check if resp is actually a http.Flusher! - how though?
+
+	// Set the headers related to event streaming.
+	ctx.Resp.Header().Set("Content-Type", "text/event-stream")
+	ctx.Resp.Header().Set("Cache-Control", "no-cache")
+	ctx.Resp.Header().Set("Connection", "keep-alive")
+	ctx.Resp.Header().Set("Access-Control-Allow-Origin", "*")
```

Comment:
I suspect that this is incorrect - but I haven't found a decent document explaining cors

Reply:
CORS headers should be user-controllable and setting `*` unconditionally can be dangerous. It seems we already have an config option, maybe use it?

https://github.com/go-gitea/gitea/blob/1bdffefc052fca440b1d4b59a5e6d402f0836374/custom/conf/app.ini.sample#L36

Reply:
I just found we have a second config option that seems to essentially do the same:

https://github.com/go-gitea/gitea/blob/1bdffefc052fca440b1d4b59a5e6d402f0836374/custom/conf/app.ini.sample#L140-L141

I don't know why we even have two options but the `cors` section certainly seems suitable to use is as it is a server-level config option, not repo-level.

Reply:
Actually I think this can be removed - it's dealt with by the Cors handler in routes.go
```suggestion
```